src/HDF5.jl

```diff
@@ -327,15 +327,15 @@ struct FixedArray{T,D,L}
     data::NTuple{L,T}
 end
 size(::Type{FixedArray{T,D,L}}) where {T,D,L} = D
-size(x::T) where {T<:FixedArray} = size(T)
+size(x::FixedArray) = size(typeof(x))
 eltype(::Type{FixedArray{T,D,L}}) where {T,D,L} = T
-eltype(x::T) where {T<:FixedArray} = eltype(T)
+eltype(x::FixedArray) = eltype(typeof(x))
 
 struct FixedString{N,PAD}
     data::NTuple{N,UInt8}
 end
 length(::Type{FixedString{N,PAD}}) where {N,PAD} = N
-length(x::T) where {T<:FixedString} = length(T)
+length(::FixedString) where {T<:FixedString} = length(T)
```

Comment:
```suggestion
length(str::FixedString) = length(typeof(str))
```

Apparently needs a test as well:
```julia
julia> fix = HDF5.FixedString{4,0}((b"test"...,))
HDF5.FixedString{4, 0}((0x74, 0x65, 0x73, 0x74))

julia> length(fix)
ERROR: UndefVarError: T not defined
Stacktrace:
 [1] length(#unused#::HDF5.FixedString{4, 0})
   @ HDF5 ~/.julia/dev/HDF5/src/HDF5.jl:338
 [2] top-level scope
   @ REPL[8]:1

julia> length(typeof(fix))
4
```

Reply:
yikes, good catch, I swear I noticed this like two weeks ago and totally forgot about it.